File: TP-1/implement/C++/utilisateur.cpp
#include <iostream>
#include <string>

class Workstation {
public:
    std::string hostname, ipAddress, os;
    
    Workstation(std::string h, std::string ip, std::string o) 
        : hostname(h), ipAddress(ip), os(o) {}
};

class User {
private:
    int id;
    std::string name, role, department;
    Workstation* workstation; // Association via un pointeur

public:
    User(int i, std::string n, std::string r, std::string d, Workstation* ws = nullptr)
        : id(i), name(n), role(r), department(d), workstation(ws) {}

    void afficherInfos() {
        std::cout << "Utilisateur: " << name;
        if (workstation) {
            std::cout << " | Poste: " << workstation->hostname << std::endl;
        } else {
            std::cout << " | Aucun poste affecte." << std::endl;
        }
    }
};

int main() {
    Workstation pcRachel("PC-SEC-01", "192.168.30.11", "Windows 10");
    User rachel(2, "Rachel", "Secretaire", "Direction", &pcRachel);

    rachel.afficherInfos();
    return 0;
}

File: TP-1/implement/python/utilisateur.py
class Workstation:
    def __init__(self, hostname, ipAddress, os):
        self.hostname = hostname
        self.ipAddress = ipAddress
        self.os = os

class User:
    def __init__(self, user_id, name, role, department, workstation=None):
        self.id = user_id
        self.name = name
        self.role = role
        self.department = department
        self.workstation = workstation

    def get_assignment_label(self):
        """Retourne une chaîne formatée pour valider l'affectation réseau"""
        # Si workstation existe, on prend son hostname, sinon on met "Aucun"
        ws_name = self.workstation.hostname if self.workstation else "Aucun"
        return f"{self.name} est affecté à {ws_name}"